File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
		int ind = logLine.indexOf(':');
		return logLine.substring(ind+1).strip();
    }

    public static String logLevel(String logLine) {
    	int find = logLine.indexOf('[');
    	int sind = logLine.indexOf(']');
    	return logLine.substring(find+1, sind).toLowerCase();
    }

    public static String reformat(String logLine) {
    	return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {

    std::string message(const char * logs){
        std::string tempstr = logs;
        int index = tempstr.find(':');
        tempstr = tempstr.substr(index+2,std::size(tempstr));
        // std::cout<<tempstr<<std::endl;
        return tempstr;
    }
    std::string log_level(const char * logs){
        std::string tempstr = logs;
        int find = tempstr.find('[');
        int lind = tempstr.find(']');
        tempstr = tempstr.substr(find+1, lind-1);
        return tempstr;
    }
    std::string reformat(const char * logs){
        std::string logLevel = log_level(logs);
        std::string messag  = message(logs);
        return messag + " (" + logLevel + ")";
    }
} // namespace log_line
